src/System.Runtime/tests/System/String.cs

```diff
@@ -1544,6 +1544,48 @@ public static void TestTrim()
 
         s = ". Foo .".TrimEnd('.');
         Assert.Equal(". Foo ", s);
+
+        s = "".Trim('c');
```

Comment:
Could we take this opportunity to clean up the tests here a little?  I think using `[Theory]` and `[InlineData]` could help make this much more readable.

Maybe split this apart into three methods?  One for each method and have cases with both a single and multiple characters to split on?

We should also continue to have some coverage for cases where an array with a single element is passed into the trim overload that takes a `char[]`